src/app/pages/BridgeWithdrawPage/components/TokenSelector/index.tsx

```diff
@@ -0,0 +1,173 @@
+    },
+    [dispatch],
+  );
+  const currentAsset = useMemo(
+    () =>
+      BridgeDictionary.get(chain as Chain, targetChain as Chain)?.getAsset(
+        sourceAsset as CrossBridgeAsset,
+      ) as AssetModel,
+    [chain, targetChain, sourceAsset],
+  );
+
+  const targetAssets = useMemo(
+    () =>
+      (
+        BridgeDictionary.get(targetChain as Chain, chain as Chain)?.assets || []
+      ).filter(item => currentAsset?.aggregatedTokens?.includes(item.asset)),
+    [chain, currentAsset, targetChain],
+  );
+
+  const [balances, setBalances] = useState<
+    { key: CrossBridgeAsset; value: string }[]
+  >([]);
+  // todo this will be used for withdrawals later.
+  useEffect(() => {
+    const callData = targetAssets.map(item => {
+      return {
+        address: (
+          currentAsset.bridgeTokenAddresses.get(item.asset) ||
+          currentAsset.tokenContractAddress
+        ).toLowerCase(),
+        abi: erc20Abi,
+        fnName: 'balanceOf',
+        args: [currentAsset.aggregatorContractAddress?.toLowerCase()],
+        key: item.symbol,
+      };
+    });
+
+    if (callData.length) {
+      bridgeNetwork
+        .multiCall(chain as any, callData)
+        .then(({ returnData }) => {
+          // todo render these balance limits to respected asset (only when chain is RSK)
+          setBalances(
+            Object.entries(returnData).map(([key, value]) => ({
+              key: key as CrossBridgeAsset,
+              value: currentAsset.fromWei((value as any).toString()) || 'x',
+            })),
+          );
+        })
+        .catch(error => {
+          console.error('what', error);
+          setBalances([]);
+        });
+    }
+  }, [chain, targetChain, targetAsset, targetAssets, currentAsset]);
+
+  const network = useMemo(
+    () => BridgeNetworkDictionary.get(targetChain as Chain),
+    [targetChain],
+  );
+
+  const getBalance = (asset: CrossBridgeAsset) =>
+    balances.find(item => item.key === asset)?.value || '0';
+
+  return (
+    <div>
+      <div className="tw-mb-20 tw-text-2xl tw-text-center tw-font-semibold">
+        Select coin to withdraw to
+      </div>
+      {targetAssets.length > 0 ? (
+        <div className="tw-flex tw-gap-10 tw-px-2 tw-justify-center">
+          {targetAssets.map(item => {
+            return (
+              <TokenItem
+                key={item.asset}
+                sourceAsset={item.asset}
+                image={item.image}
+                symbol={item.symbol}
+                balance={getBalance(item.asset)}
+                onClick={() => selectTargetAsset(item.asset)}
+              />
+            );
+          })}
+        </div>
+      ) : (
+        <p>
+          Sorry, no supported withdraw tokens for {sourceAsset} on{' '}
+          {network?.name} network. Try choosing another network or token.
+        </p>
+      )}
+    </div>
+  );
+}
+
+function TokenItem({ sourceAsset, balance, image, symbol, onClick }) {
```

Comment:
Extract to a separate file. Also, cannot we just reuse `TokenItem` from `BridgeDepositPage`?